### ISSUE TITLE
Stats randomly crashes in the background. 

### ISSUE THREAD

**Issue by coldenate** [exelban/stats#1402]
Hi! I have noticed stats will just disappear from my menubar and I will have to restart it. What is the process to take to give you the logs?

**Comment by exelban** [comment]
Do you have the latest version installed?

**Comment by coldenate** [comment]
I do. I am currently waiting to see if the behavior repeats. It may be a false positive since I cannot tell when it is updated. I had this issue yesterday mostly.

**Comment by exelban** [comment]
Please check what exactly version of Stats you have

**Comment by coldenate** [comment]
Version 2.8.16

**Comment by coldenate** [comment]
I believe updating fixed the issue. I will let you know if I see it happen again.

**Comment by exelban** [comment]
ok, give me to know.

**Comment by exelban** [comment]
looks like it's fixed

**Comment by bemfica** [comment]
Not sure if it's related to this, but it was the closest issue I found with what I'm experiencing. Stats sometimes just closes unexpectedly, no crash window, it just disappears from the menu bar. Tried disabling storage widgets as suggested in other issue, didn't help. Starting Stats from the terminal gives me the following message when the modules disappear:
`Object` 0x[RANDOM HEXADECIMAL] `of class _ContiguousArrayStorage deallocated with non-zero retain count 2. This object's deinit, or something called from it, may have created a strong reference to self which outlived deinit, resulting in a dangling reference.`
